runtime/lua/vim/treesitter/query.lua

```diff
@@ -528,7 +531,7 @@ local directive_handlers = {
 
     while true do
```

Comment:
`while true` makes me nervous. Can't we change this to `while end_row >= start_row`?